sympy/integrals/prde.py

```diff
@@ -755,56 +755,65 @@ def limited_integrate_reduce(fa, fd, G, DE):
 # `i = 0` to `i = m` where c0 = 1
 # and ci = -ci for i = 1 to m.
 
-# given an input for the `limited_integrate` functon
-def interpret_lim_int_as_param_rischDE(fa, fd, G, DE):
-    # f = Fa/Fd
-    Fa = Poly(0, DE.t)
-    # Fd can be set to any non-zero Poly
-    Fd = Poly(1, DE.t)
-    G = [(fa, fd)] + G
-    return Fa, Fd, G, DE
-
-
 def limited_integrate(fa, fd, G, DE):
     """
     Solves the limited integration problem:  f = Dv + Sum(ci*wi, (i, 1, n))
     """
     fa, fd = fa*Poly(1/fd.LC(), DE.t), fd.monic()
-    A, B, h, N, g, V = limited_integrate_reduce(fa, fd, G, DE)
-    V = [g] + V
-    g = A.gcd(B)
-    A, B, V = A.quo(g), B.quo(g), [via.cancel(vid*g, include=True) for
-        via, vid in V]
-    Q, M = prde_linear_constraints(A, B, V, DE)
-    M, _ = constant_system(M, zeros(M.rows, 1), DE)
-    l = M.nullspace()
-    if M == Matrix() or len(l) > 1:
-        # Continue with param_rischDE()
-        Fa, Fd, G, DE = interpret_lim_int_as_param_rischDE(fa, fd, G, DE)
-        h, A = param_rischDE(Fa, Fd, G, DE)
-        r, m = len(h), len(W) - r
-        W = A.nullspace()
-        alpha = W[0]
-        C = [1 if i == 0 else -W[i]*alpha for i in range(m)]
-        y = sum([Poly(h[i][0], DE.t)/Poly(h[i][1], DE.t)*W[m + i] for i in range(r)])
-        return y, C
-    elif len(l) == 0:
-        raise NonElementaryIntegralException
-    elif len(l) == 1:
-        # The c1 == 1.  In this case, we can assume a normal Risch DE
-        if l[0][0].is_zero:
-            raise NonElementaryIntegralException
-        else:
-            l[0] *= 1/l[0][0]
-            C = sum([Poly(i, DE.t)*q for (i, q) in zip(l[0], Q)])
-            # Custom version of rischDE() that uses the already computed
-            # denominator and degree bound from above.
-            B, C, m, alpha, beta = spde(A, B, C, N, DE)
-            y = solve_poly_rde(B, C, m, DE)
-
-            return ((alpha*y + beta, h), list(l[0][1:]))
+    # interpretting limited integration problem as a
+    # parametric Risch DE problem
+    Fa = Poly(0, DE.t)
+    Fd = Poly(1, DE.t)
+    G = [(fa, fd)] + G
+    h, A = param_rischDE(Fa, Fd, G, DE)
+    V = A.nullspace()
+    V = [v for v in V if v[0] != 0]
+    if V is None:
+        return None
```

Comment:
We may eventually want to handle also nonelementary functions. `limited_integrate` may then be called from different places in the code, and it should be up to the caller to decide whether to raise an exception or do something else. We should consider `limited_integrate` as a low level routine that only answers the question whether there is an integral in the limited sense or not.